engine/common/rpc/convert/convert.go

```diff
@@ -814,3 +816,147 @@ func MessageToChunk(m *entities.Chunk) (*flow.Chunk, error) {
 		EndState:  endState,
 	}, nil
 }
+
+func BlockExecutionDataToMessage(data *execution_data.BlockExecutionData) (*entities.BlockExecutionData, error) {
```

Comment:
please add tests converting to/from messages to ensure the conversion is correct